src/pudl/helpers.py

```diff
@@ -2185,3 +2185,305 @@ def retry(
+                        print(df[df[col].isin(custom_missing_values)].head())
+                    if na_count == 0 and custom_missing_count == 0:
+                        print("Found nothing worth reporting here")
+                else:
+                    print(f"Column '{col}' is datetime but contains only NA values.")
+
+            # If the column is of some other type, simply note the type
+            else:
+                print(f"Column '{col}' is of type {col_data.dtype}.")
+            print("")
+
+        except Exception as e:
+            exception_cols.append(col)
+            print(f"Caught exception for column {col}: {e}\n")
+            continue
+
+    print(f"Columns with NaNs or custom missing values: {nan_cols}")
+    print(f"Columns with exceptions during processing: {exception_cols}")
+
+    return exception_cols
+
+
+def standardize_state_columns(df: pd.DataFrame, state_columns: list) -> pd.DataFrame:
+    """Standardizes the state strings in the specified columns by replacing full state names
+    with their two-letter abbreviations and capitalizing the strings.
+
+    Args:
+    df: The DataFrame containing the state columns to standardize.
+    state_columns: A list of column names where state standardization should be applied.
+
+    Returns:
+    The DataFrame with standardized state columns.
+    """
+    # Dictionary mapping full state names to their two-letter abbreviations
+    state_abbreviations = {
+        "Alabama": "AL",
+        "Alaska": "AK",
+        "Arizona": "AZ",
+        "Arkansas": "AR",
+        "California": "CA",
+        "Colorado": "CO",
+        "Connecticut": "CT",
+        "Delaware": "DE",
+        "Florida": "FL",
+        "Georgia": "GA",
+        "Hawaii": "HI",
+        "Idaho": "ID",
+        "Illinois": "IL",
+        "Indiana": "IN",
+        "Iowa": "IA",
+        "Kansas": "KS",
+        "Kentucky": "KY",
+        "Louisiana": "LA",
+        "Maine": "ME",
+        "Maryland": "MD",
+        "Massachusetts": "MA",
+        "Michigan": "MI",
+        "Minnesota": "MN",
+        "Mississippi": "MS",
+        "Missouri": "MO",
+        "Montana": "MT",
+        "Nebraska": "NE",
+        "Nevada": "NV",
+        "New Hampshire": "NH",
+        "New Jersey": "NJ",
+        "New Mexico": "NM",
+        "New York": "NY",
+        "North Carolina": "NC",
+        "North Dakota": "ND",
+        "Ohio": "OH",
+        "Oklahoma": "OK",
+        "Oregon": "OR",
+        "Pennsylvania": "PA",
+        "Rhode Island": "RI",
+        "South Carolina": "SC",
+        "South Dakota": "SD",
+        "Tennessee": "TN",
+        "Texas": "TX",
+        "Utah": "UT",
+        "Vermont": "VT",
+        "Virginia": "VA",
+        "Washington": "WA",
+        "West Virginia": "WV",
+        "Wisconsin": "WI",
+        "Wyoming": "WY",
+    }
+
+    for col in state_columns:
+        df[col] = df[col].replace(state_abbreviations).str.upper()
```

Comment:
When you have a complete mapping between old and new values, the [col.map()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.map.html) method should be much faster. However any value that appears in the series but not in the dictionary defining the mapping gets converted to NA.

Reply:
Updated logic for standardizing state abbreviations here:

https://github.com/catalyst-cooperative/pudl/pull/3929/files#diff-941a3989c384fa527b5188c725a5b3beb1f1e0dbf9d427dae9dd82f4f9fe5c09R126